devito/types/basic.py

```diff
@@ -1203,3 +1203,37 @@ def origin(self):
     def free_symbols(self):
         # Make it cached, since it's relatively expensive and called often
         return super(Indexed, self).free_symbols
+
+    def compare(self, other):
+        """
+        Override `sympy.Basic.compare` to honor Devito's canonical ordering
+        of arguments.
+        In SymPy:
+
+            f[x+1] < f[x+2] < ... < f[x+9] < f[x]
+
+        While in Devito we pretend
+
+            f[x] < f[x+1] < f[x+2] < ... < f[x+9]
+
+        That is the arguments need to be ordered monothonically based on the indices
+        so that the symbolic trees of two derivative expressions can be compared
+        argument-wise.
+        """
+        if (self.__class__ != other.__class__) or (self.function is not other.function):
+            return super().compare(other)
+        for l, r in zip(self.indices, other.indices):
+            try:
+                expr = l - r
+                if expr > 0:
+                    c = 1
+                elif expr == 0:
+                    c = 0
+                else:
+                    c = -1
```

Comment:
how is that different than `c = sign(l-r)`?

Reply:
oops, indeed, good catch. Using `sign` now